internal/api/option/auth.go

```diff
@@ -243,6 +256,46 @@ func (a *authInterceptor) WrapStreamingHandler(
 		return next(ctx, conn)
 	}
 }
+func (a *authInterceptor) listServiceAccounts(
+	ctx context.Context,
+	username string,
+	groups []string,
+) (map[string][]types.NamespacedName, error) {
+	queries := []libClient.MatchingFields{
+		{
+			kubeclient.ServiceAccountsBySubjectIndexField: username,
+		},
+	}
+	for _, group := range groups {
+		queries = append(queries, libClient.MatchingFields{
+			kubeclient.ServiceAccountsByGroupIndexField: group,
+		})
+	}
+	accounts := make(map[string]map[types.NamespacedName]struct{})
+	for _, q := range queries {
+		list := &corev1.ServiceAccountList{}
+		if err := a.internalClient.List(ctx, list, q); err != nil {
+			return nil, errors.Wrap(err, "list service accounts")
+		}
+		for _, sa := range list.Items {
+			key := types.NamespacedName{
+				Namespace: sa.GetNamespace(),
+				Name:      sa.GetName(),
+			}
+			if _, ok := accounts[key.Namespace]; !ok {
+				accounts[key.Namespace] = make(map[types.NamespacedName]struct{})
+			}
+			accounts[key.Namespace][key] = struct{}{}
+		}
+	}
+	res := make(map[string][]types.NamespacedName)
+	for ns, names := range accounts {
+		for name := range names {
+			res[ns] = append(res[ns], name)
+		}
+	}
```

Comment:
How much do we really need this loop?

Can we just return the `map[string]map[types.NamespacedName]struct{}` instead of returning `map[string][]types.NamespacedName`?

I think that would be ok, because I don't think order really matters when we use this information for authz decisions later on.

wdyt?